modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/deployers/terraform/TerraformBootDeployment.java

```diff
@@ -0,0 +1,203 @@
+        return System.getProperty("java.io.tmpdir")
+                + File.separator + terraformLocalConfig.getWorkspaceDirectory() + File.separator
+                + taskId;
+    }
+
+    private TerraformAsyncDestroyFromDirectoryRequest getDestroyRequest(DeployTask task,
+            String stateFile)
+            throws UnknownHostException {
+        TerraformAsyncDestroyFromDirectoryRequest request =
+                new TerraformAsyncDestroyFromDirectoryRequest();
+        request.setScripts(getFiles(task));
+        request.setTfState(stateFile);
+        request.setVariables(getInputVariables(task));
+        request.setEnvVariables(getEnvironmentVariables(task));
+        WebhookConfig hookConfig = new WebhookConfig();
+        String callbackUrl = terraformBootConfig.getClientRequestBaseUrl()
+                + terraformBootConfig.getDestroyCallbackUri();
+        hookConfig.setUrl(callbackUrl + task.getId());
+        hookConfig.setAuthType(AuthTypeEnum.NONE);
+        request.setWebhookConfig(hookConfig);
+        return request;
+    }
+
+    private TerraformAsyncDeployFromDirectoryRequest getDeployRequest(DeployTask task)
+            throws UnknownHostException {
+        TerraformAsyncDeployFromDirectoryRequest request =
+                new TerraformAsyncDeployFromDirectoryRequest();
+        request.setIsPlanOnly(false);
+        request.setScripts(getFiles(task));
+        request.setVariables(getInputVariables(task));
+        request.setEnvVariables(getEnvironmentVariables(task));
+        WebhookConfig hookConfig = new WebhookConfig();
+        String callbackUrl = terraformBootConfig.getClientRequestBaseUrl()
+                + terraformBootConfig.getDeployCallbackUri();
+        hookConfig.setUrl(callbackUrl + task.getId());
+        hookConfig.setAuthType(AuthTypeEnum.NONE);
+        request.setWebhookConfig(hookConfig);
+        return request;
+    }
+
+    private List<String> getFiles(DeployTask task) {
+        Csp csp = task.getCreateRequest().getCsp();
+        String version = terraformVersionProvider.getTerraformVersionByCsp(csp);
+        String region = task.getCreateRequest().getRegion();
+        String provider = TerraformProviders.getProvider(csp).getProvider(version, region);
+        String deployer = task.getOcl().getDeployment().getDeployer();
+        return Arrays.asList(provider, deployer);
+    }
+
+    private Map<String, Object> getInputVariables(DeployTask deployTask) {
+        Map<String, Object> inputVariables = new HashMap<>();
+        inputVariables.putAll(this.deployEnvironments.getVariables(deployTask));
+        inputVariables.putAll(this.deployEnvironments.getFlavorVariables(deployTask));
+        return inputVariables;
+    }
+
+    private Map<String, String> getEnvironmentVariables(DeployTask deployTask) {
+        Map<String, String> envVariables = this.deployEnvironments.getEnv(deployTask);
+        if (terraformLocalConfig.isDebugEnabled()) {
```

Comment:
we do not need this. We will not control the terraform-boot log level from xpanse